ntirpc/rpc/gss_internal.h

```diff
@@ -113,30 +112,17 @@ svc_rpc_gss_data *alloc_svc_rpc_gss_data(void)
 }
 
 static inline void
-unref_svc_rpc_gss_data(struct svc_rpc_gss_data *gd,
-		       uint32_t flags)
+unref_svc_rpc_gss_data(struct svc_rpc_gss_data *gd)
 {
-	u_int refcnt;
-	bool gd_locked = flags & SVC_RPC_GSS_FLAG_LOCKED;
-
-	refcnt = atomic_dec_uint32_t(&gd->refcnt);
+	mutex_lock(&gd->lock);
```

Comment:
the thread can access the gd field with its own references. It get it from rec, as stated above and inc the refcount again. Since there's no lock preventing this it would succeed.
Anyhow, I'm not introducing the lock, it was always there, maybe not always pulled but therefore we got rid of a few other commands and a simple mutex is cheap.